ql/src/java/org/apache/hadoop/hive/ql/plan/ReduceSinkDesc.java

```diff
@@ -141,6 +141,9 @@ private ReducerTraits(int trait) {
   // used to decide whether global order is needed
   private transient boolean hasOrderBy = false;
 
+  // used to decide whether topn key optimisation can be applied
+  private transient boolean hasOnlyOrderByLimit = false;
```

Comment:
Do we really need this new indicator here? IT seems to be a hint about the structure of the plan rather than a property/quality of the `ReduceSink` operator.